sunspec/core/util.py

```diff
@@ -95,6 +96,12 @@ def data_to_double(data):
         return d[0]
 
 def data_to_str(data):
+
+    # Change the data from bytes string to regular string for python 3
+    # compatibility
+    if sys.version_info > (3,):
+        data = str(data, 'latin-1')
```

Comment:
This may actually be a specification issue. 12041 doesn't actually define an encoding.  Either the specification should define an encoding (if so I'd figure UTF-8)  or this library shouldn't encode/decode because it can't know what the encoding is.  If the latter, the interface should just be bytes and the application must handle the encoding/decoding.  In that case a note in the spec would probably be good making this explicit, that the 'string' type is really bytes.

Also, encoding and decoding are still relevant in py2 since the meaning of a given set of bytes still varies depending on the coding used.  https://repl.it/JRTE

Also also, we should be using the compatibility libraries that provide str and bytes as unique types in py2.  :]